src/cpp/rtps/xmlparser/XMLElementParser.cpp

```diff
@@ -4715,44 +4716,208 @@ XMLP_ret XMLParser::getXMLEntityFactoryQos(
 XMLP_ret XMLParser::getXMLBuiltinTransports(
         tinyxml2::XMLElement* elem,
         eprosima::fastdds::rtps::BuiltinTransports* bt,
-        uint8_t /*ident*/)
+        eprosima::fastdds::rtps::BuiltinTransportsOptions* bt_opts,
+        uint8_t ident)
 {
     /*
-        <xs:simpleType name="builtinTransports">
-            <xs:restriction base="xs:string">
-                <xs:enumeration value="NONE"/>
-                <xs:enumeration value="DEFAULT"/>
-                <xs:enumeration value="DEFAULTv6"/>
-                <xs:enumeration value="SHM"/>
-                <xs:enumeration value="UDPv4"/>
-                <xs:enumeration value="UDPv6"/>
-                <xs:enumeration value="LARGE_DATA"/>
-                <xs:enumeration value="LARGE_DATAv6"/>
-            </xs:restriction>
-        </xs:simpleType>
+        <xs:complexType name="builtinTransportsEnumType">
+            <xs:all>
+                <xs:element name="mode" minOccurs="1" maxOccurs="1">
+                    <xs:simpleType>
+                        <xs:restriction base="xs:string">
+                            <xs:enumeration value="NONE" />
+                            <xs:enumeration value="DEFAULT" />
+                            <xs:enumeration value="DEFAULTv6" />
+                            <xs:enumeration value="SHM" />
+                            <xs:enumeration value="UDPv4" />
+                            <xs:enumeration value="UDPv6" />
+                            <xs:enumeration value="LARGE_DATA" />
+                            <xs:enumeration value="LARGE_DATAv6" />
+                        </xs:restriction>
+                    </xs:simpleType>
+                </xs:element>
+            </xs:all>
+            <xs:attribute name="max_msg_size" type="string" use="optional"/>
+            <xs:attribute name="sockets_size" type="string" use="optional"/>
+            <xs:attribute name="non_blocking" type="string" use="optional"/>
+        </xs:complexType>
```

Comment:
Update with latest changes
```suggestion
    <xs:simpleType name="builtinTransportKind">
        <xs:restriction base="xs:string">
            <xs:enumeration value="NONE" />
            <xs:enumeration value="DEFAULT" />
            <xs:enumeration value="DEFAULTv6" />
            <xs:enumeration value="SHM" />
            <xs:enumeration value="UDPv4" />
            <xs:enumeration value="UDPv6" />
            <xs:enumeration value="LARGE_DATA" />
            <xs:enumeration value="LARGE_DATAv6" />
        </xs:restriction>
    </xs:simpleType>
    
    <xs:complexType name="builtinTransportsType" >
        <xs:simpleContent>
            <xs:extension base="builtinTransportKind">
                <xs:attribute name="max_msg_size" type="string" use="optional"/>
                <xs:attribute name="sockets_size" type="string" use="optional"/>
                <xs:attribute name="non_blocking" type="string" use="optional"/>
                <xs:attribute name="tcp_negotiation_timeout" type="string" use="optional"/>
            </xs:extension>
        </xs:simpleContent>
    </xs:complexType>
```

Reply:
Update the types as the previous suggestion if considered
```suggestion
    <xs:simpleType name="builtinTransportKind">
        <xs:restriction base="xs:string">
            <xs:enumeration value="NONE" />
            <xs:enumeration value="DEFAULT" />
            <xs:enumeration value="DEFAULTv6" />
            <xs:enumeration value="SHM" />
            <xs:enumeration value="UDPv4" />
            <xs:enumeration value="UDPv6" />
            <xs:enumeration value="LARGE_DATA" />
            <xs:enumeration value="LARGE_DATAv6" />
        </xs:restriction>
    </xs:simpleType>
    
    <xs:complexType name="builtinTransportsType" >
        <xs:simpleContent>
            <xs:extension base="builtinTransportKind">
                <xs:attribute name="max_msg_size" type="uint32" use="optional"/>
                <xs:attribute name="sockets_size" type="uint32" use="optional"/>
                <xs:attribute name="non_blocking" type="boolean" use="optional"/>
                <xs:attribute name="tcp_negotiation_timeout" type="uint32" use="optional"/>
            </xs:extension>
        </xs:simpleContent>
    </xs:complexType>
```